husc/screens/cellomics.py

```diff
@@ -0,0 +1,263 @@
+        new_fn = ''.join(new_fn)
+        stitched_image = run_snail_stitch(value)
+        io.imsave(os.path.join(dir, new_fn), rescale_12bit(stitched_image))
+
+
+def rescale_12bit(image, bit='8'):
+    """Rescale a 12bit image.
+
+    Cellomics TIFs ar encoded as 12bit TIF files, which generally cannot
+    be viewed in most software. This function rescales the images to either an
+    8 or 16 bit TIF file.
+
+    Parameters
+    ----------
+    image : array, shape (M, N)
+        The image to be rescaled.
+
+    Returns
+    -------
+    scale_image : array, shape (M, N)
+        The rescaled image.
+    """
+    if bit == '8':
+        scale_image = np.round((image/4095.) * 255).astype(np.uint8)
+    elif bit == '16':
+        scale_image = np.round((image/4095.) * 65535).astype(np.uint16)
+    else:
+        scale_image = np.round(image/4095.).astype(np.float)
+    return scale_image
+
+
+def run_snail_stitch(fns):
+    """Run right, clockwise spiral/snail stitching of 25 Cellomics TIFs.
+
+    Runs clockwise stitching of the images. The spiral begins in the
+    center of the image, moves to the right and circles around in a clockwise
+    manner.
+
+    Parameters
+    ----------
+    fns : list of array, shape (M, N)
+        The list of 25 image files to be stitched together.
+
+    Returns
+    -------
+    stitched_image : array, shape (5*M, 5*N)
+        The stitched image.
+    """
+    right = [[20, 21, 22, 23, 24],
+             [19, 6, 7, 8, 9],
+             [18, 5, 0, 1, 10],
+             [17, 4, 3, 2, 11],
+             [16, 15, 14, 13, 12]]
+
+    stitched_image = np.array([])
+    for i in range(0, 5):
+        stitched_row = np.array([])
+        for j in range(0, 5):
+            image = io.imread(fns[right[i][j]])
+            stitched_row = concatenate(stitched_row, image)
+        stitched_image = stack(stitched_image, stitched_row)
+    return stitched_image
+
+
+def stack(arr1, arr2):
+    """No docstring here as this function wil be removed soon.
+    """
+    if arr1.shape[0] == 0:
+        return arr2
+    else:
+        return np.vstack((arr1, arr2))
+
+
+def concatenate(arr1, arr2):
+    """No docstring here as this function wil be removed soon.
+    """
+    if arr1.shape[0] == 0:
+        return arr2
+    else:
+        return np.concatenate((arr1, arr2), 1)
+
+
+def make_wellchannel2file_dict(fns):
+    """Return a dictionary mapping well co-ordinates to filenames.
+
+    Returns a dictionary where key are (plate, well) co-ordinates and
+    values are lists of images corresponding to that plate and well.
+
+    Parameters
+    ----------
+    fns : list of string
+        A list of Cellomics TIF files.
+
+    Returns
+    -------
+    wellchannel2file : dict {tuple : list of string}
+        The dictionary mapping the (plate, well) co-ordinate to
+        a list of files corresponding to that well.
+    """
+    wellchannel2file = coll.defaultdict(list)
+    for fn in fns:
```

Comment:
Run a CyToolz.groupby call here instead of operating.

Additionally, since we've decided to unify wellchannel as an ID across all screens, I suggest we use a common, simpler name across all functions. e.g. make_key2file_dict. Or just: make_key2file(fns). ;)